### PR TITLE
[docs] Framework docs

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

- Update to Docusaurus 3
- Add components to global scope so they don't need to be imported in mdx files.
- Put framework docs into site
- Fix mdx docs to build properly under d3
- Add plugin to put descriptions into globaldata
- Put framework docs into move section

## Test Plan 

The PR boils down to what's shown at https://sui-core-git-fork-ronny-mysten-frameworkdocs-sui-foundation.vercel.app/references/framework

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-framework/build.rs
Comment (ronny-mysten) [inline crates/sui-framework/build.rs:1]: @tzakian Doubt this is an issue but want to make sure.

Comment (tzakian) [reply crates/sui-framework/build.rs:1]: Not an issue at all! Change LGTM :)

File: docs/site/src/theme/Navbar/ColorModeToggle/styles.module.css [hunk 1]
```diff
@@ -3,4 +3,4 @@
 }
 .darkNavbarColorModeToggle:hover {
   background: var(--ifm-color-gray-800);
-}
+}
```
Comment (tnowacki) [inline docs/site/src/theme/Navbar/ColorModeToggle/styles.module.css:6]: 👀

File: docs/site/src/theme/Navbar/Search/styles.module.css [hunk 2]
```diff
@@ -12,4 +27,4 @@
     display: flex;
     column-gap: 1rem;
   }
-}
+}
```
Comment (tnowacki) [inline docs/site/src/theme/Navbar/Search/styles.module.css:30]: 👀

**Review by tnowacki — APPROVED** [review]

File: docs/content/concepts/graphql-rpc.mdx [hunk 3]
```diff
@@ -202,7 +202,7 @@ Fields that return a paginated response accept at least the following optional p
 
 They also return a type that conforms to the [GraphQL Cursor Connections Specification](https://relay.dev/graphql/connections.htm), meaning its name ends in `Connection`, and it contains at least the following fields:
 
-- `pageInfo`, of type [PageInfo](../references/sui-api/sui-graphql/reference/objects/page-info.mdx), which indicates whether there are more pages before or after the page returned.
+- `pageInfo`, of type <a href="/references/sui-api/sui-graphql/reference/objects/page-info" data-noBrokenLinkCheck='true'>PageInfo</a>, which indicates whether there are more pages before or after the page returned.
```
Comment (stefan-mysten) [inline docs/content/concepts/graphql-rpc.mdx:205]: Out of curiosity, this is now the recommended way to point to a page inside our docs?

File: docs/content/guides/developer/getting-started/get-coins.mdx [hunk 4]
```diff
@@ -9,7 +9,7 @@ Sui faucet is a helpful tool where Sui developers can get free test SUI tokens t
 To request tokens from the faucet, you must own a wallet address that can receive the SUI tokens. See the [Get Sui Address](./get-address.mdx) topic if you don't already have an address or need to create a new one.
 
 ## Request test tokens via the CLI 
-If you are using the Devnet or Testnet networks, or you spun up a local network, you can use the [Sui CLI](https://docs.sui.io/guides/developer/getting-started/sui-install) to request tokens for your address. The `sui client faucet` uses the [active network](references/cli/client#list-available-network-environments) and [active address](references/cli/client#get-current-active-address) that is currently set in the Sui CLI by default, but you can specify custom data through the following two arguments:
+If you are using the Devnet or Testnet networks, or you spun up a local network, you can use the [Sui CLI](../../../references/cli/client.mdx) to request tokens for your address. The `sui client faucet` uses the active network and active address that is currently set in the Sui CLI by default, but you can specify custom data through the following two arguments:
```
Comment (stefan-mysten) [inline docs/content/guides/developer/getting-started/get-coins.mdx:12]: See my question above on passing links to other pages.

**Review by amnn — APPROVED** [review]
I think this is good to go, but couple of things we can follow-up on quickly (e.g. before declaring victory):

- I think we've gone too far with eliminating addresses. Can we eliminate them from the sidebar without eliminating them from the titles in each page? (Alternatively, we could add them in the sidebar, but just on the overall title, e.g. "0x1: Move Stdlib" or similar).
- We are currently missing Sui System and Deepbook as well -- where did they go?

File: docs/content/sidebars/references.js [hunk 5]
```diff
@@ -101,6 +96,15 @@ const references = [
...
 		items: [
+			{
+				type: 'category',
+				label: 'Framework',
+				link: {
+					type: 'doc',
+					id: 'references/framework',
+				},
+				items: [{ type: 'autogenerated', dirName: 'references/framework' }],
+			},
```
Comment (stefan-mysten) [inline docs/content/sidebars/references.js:107]: formatting seems a bit off.

Comment (stefan-mysten) [reply docs/content/sidebars/references.js:107]: Also, we probably want spaces instead of tabs here.

File: docs/site/docusaurus.config.js [hunk 6]
```diff
@@ -1,26 +1,35 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-const lightCodeTheme = require("prism-react-renderer/themes/github");
-const darkCodeTheme = require("prism-react-renderer/themes/nightOwl");
-const math = require("remark-math");
-const katex = require("rehype-katex");
+import { themes } from "prism-react-renderer";
```
Comment (stefan-mysten) [inline docs/site/docusaurus.config.js:4]: Don't we also need the darkCodeTheme? Or the `themes` now includes both and we can later interchange?

File: docs/site/docusaurus.config.js [hunk 7]
```diff
@@ -1,26 +1,35 @@
...
+  // Set the /<baseUrl>/ pathname under which your site is served
+  // For GitHub pages deployment, it is often '/<projectName>/'
   baseUrl: "/",
   customFields: {
     amplitudeKey: process.env.AMPLITUDE_KEY,
   },
+
   onBrokenLinks: "throw",
-  onBrokenMarkdownLinks: "throw",
+  onBrokenMarkdownLinks: "warn",
```
Comment (stefan-mysten) [inline docs/site/docusaurus.config.js:28]: I assume these warnings are at build time, right?

File: docs/site/docusaurus.config.js [hunk 8]
```diff
@@ -213,17 +224,12 @@ const config = {
...
-
-      /**
-       * Syntax Highlighting Configuration
-       * TODO: add better themes like Atom One Dark / Light
-       */
       prism: {
-        theme: lightCodeTheme,
-        darkTheme: darkCodeTheme,
+        theme: themes.github,
+        darkTheme: themes.nightOwl,
```
Comment (stefan-mysten) [inline docs/site/docusaurus.config.js:229]: I see, we can now simply call the right theme through their fields.

**Review by stefan-mysten — APPROVED** [review]
Awesome work @ronny-mysten!!

File: docs/content/concepts/graphql-rpc.mdx (hunk 3 above)
Comment (ronny-mysten) [reply docs/content/concepts/graphql-rpc.mdx:205]: No. It's the way we must point to docs now that are automatically generated.

**Pull request merged**